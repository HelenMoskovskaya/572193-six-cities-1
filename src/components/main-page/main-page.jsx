import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import OfferList from '../offer-list/offer-list.jsx';
import MapCity from '../map/map.jsx';
import CitiesList from '../cities-list/cities-list.jsx';
import Header from '../header/header.jsx';
import Svg from '../svg/svg.jsx';
import {connect} from 'react-redux';
import {getActiveCity, getCityList, getActiveOffers, getLoadStatus} from '../../reducer/data/selectors.js';
import {getAuthorizationStatus, getUserData} from '../../reducer/user/selectors.js';
import {ActionCreatorData} from '../../reducer/data/data.js';
import withActiveItem from '../../hocs/with-active-item.jsx';
import {compose} from 'recompose';


class MainPage extends PureComponent {
  constructor(props) {
    super(props);
  }

  render() {
    const {offers, cities, onCityClick, city, isAuthorizationRequired, userData, isLoadOffers, handleActivateElement, activeItem} = this.props;
    if (isLoadOffers) {
      return <div className="page page--gray page--main">

        <Svg />
        <Header isAuthorizationRequired = {isAuthorizationRequired} userData = {userData}/>

        <main className="page__main page__main--index">
          <h1 className="visually-hidden">
            Cities</h1>

          <CitiesList cities={cities} city={city} onCityClick={(activeCity) => {
            onCityClick(activeCity);
            handleActivateElement(null);
          }}/>

          <div className="cities__places-wrapper">
            <div className="cities__places-container container">
              <section className="cities__places places">
                <h2 className="visually-hidden">Places</h2>
                <b className="places__found">{`${offers.length} ${offers.length === 1 ? `place` : `places`} 
                to stay in ${city.name}`}</b>
                <OfferList offers={offers} handleActivateElement= {handleActivateElement}
                />
              </section>
              <div className="cities__right-section">
                <section className="cities__map map">

                  <MapCity offers={offers}
                    city={city}
                    activeOffer={activeItem}/>

                </section>
              </div>
            </div>
          </div>
        </main>
      </div>;
    }
    return null;
  }
}

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  city: getActiveCity(state),
  offers: getActiveOffers(state),
  cities: getCityList(state),
  isAuthorizationRequired: getAuthorizationStatus(state),
  userData: getUserData(state),
  isLoadOffers: getLoadStatus(state),
});

const mapDispatchToProps = (dispatch) => ({
  onCityClick: (activeCity) => {
    dispatch(ActionCreatorData.changeCity(activeCity));
  },
});


MainPage.propTypes = {
  offers: PropTypes.arrayOf(PropTypes.shape({
    city: PropTypes.shape({
      name: PropTypes.string.isRequired,
      location: PropTypes.shape({
        latitude: PropTypes.number.isRequired,
        longitude: PropTypes.number.isRequired,
        zoom: PropTypes.number.isRequired
      })
    }),
    previewImage: PropTypes.string.isRequired,
    images: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    title: PropTypes.string.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    isPremium: PropTypes.bool.isRequired,
    rating: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    bedrooms: PropTypes.number.isRequired,
    maxAdults: PropTypes.number.isRequired,
    price: PropTypes.number.isRequired,
    goods: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
    host: PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      isPro: PropTypes.bool.isRequired,
      avatarUrl: PropTypes.string.isRequired
    }),
    description: PropTypes.string.isRequired,
    location: PropTypes.shape({
      latitude: PropTypes.number.isRequired,
      longitude: PropTypes.number.isRequired,
      zoom: PropTypes.number.isRequired
    }),
    id: PropTypes.number.isRequired
  })).isRequired,

  city: PropTypes.object.isRequired,
  cities: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  onCityClick: PropTypes.func.isRequired,
  isAuthorizationRequired: PropTypes.bool.isRequired,
  userData: PropTypes.object,
  isLoadOffers: PropTypes.bool.isRequired,
  handleActivateElement: PropTypes.func.isRequired,
  activeItem: PropTypes.object
};

export {MainPage};
export default compose(
    connect(mapStateToProps, mapDispatchToProps),
    withActiveItem
)(MainPage);
