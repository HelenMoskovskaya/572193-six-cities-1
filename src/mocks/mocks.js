export const mockOffers = [
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 51.225402,
        longitude: 6.776314,
        zoom: 13
      }
    },
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
    images: [
      `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`
    ],
    title: `Waterfront with extraordinary view`,
    isFavorite: false,
    isPremium: true,
    rating: 2.7,
    type: `hotel`,
    bedrooms: 2,
    maxAdults: 8,
    price: 393,
    goods: [
      `Breakfast`,
      `Washer`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    location: {
      latitude: 51.214402,
      longitude: 6.764314000000001,
      zoom: 16
    },
    id: 1
  },
  {
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 51.225402,
        longitude: 6.776314,
        zoom: 13
      }
    },
    previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
    images: [
      `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
      `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`
    ],
    title: `Waterfront with extraordinary view`,
    isFavorite: false,
    isPremium: true,
    rating: 2.7,
    type: `hotel`,
    bedrooms: 2,
    maxAdults: 8,
    price: 393,
    goods: [
      `Breakfast`,
      `Washer`,
      `Laptop friendly workspace`
    ],
    host: {
      id: 25,
      name: `Angelina`,
      isPro: true,
      avatarUrl: `img/avatar-angelina.jpg`
    },
    description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
    location: {
      latitude: 51.214402,
      longitude: 6.764314000000001,
      zoom: 16
    },
    id: 3
  }
];

export const mockOffer = {
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 51.225402,
      longitude: 6.776314,
      zoom: 13
    }
  },
  previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
  images: [
    `https://es31-server.appspot.com/six-cities/static/hotel/18.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/12.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/20.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/17.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/15.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/2.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/8.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/5.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/6.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/1.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/7.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`,
    `https://es31-server.appspot.com/six-cities/static/hotel/14.jpg`
  ],
  title: `Waterfront with extraordinary view`,
  isFavorite: false,
  isPremium: true,
  rating: 2.7,
  type: `hotel`,
  bedrooms: 2,
  maxAdults: 8,
  price: 393,
  goods: [
    `Breakfast`,
    `Washer`,
    `Laptop friendly workspace`
  ],
  host: {
    id: 25,
    name: `Angelina`,
    isPro: true,
    avatarUrl: `img/avatar-angelina.jpg`
  },
  description: `I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!`,
  location: {
    latitude: 51.214402,
    longitude: 6.764314000000001,
    zoom: 16
  },
  id: 1
};

export const cityMock = {
  name: `Amsterdam`,
  location: {
    latitude: 51.225402,
    longitude: 6.776314,
    zoom: 13
  }
};

export const cities = [`Amsterdam`, `Paris`, `Colone`];

export const userData = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`,
  isPro: false
};

export const mockFavorites = {
  "Amsterdam": mockOffers,
  "Colone": mockOffers
};

export const reviewsMock = [
  {
    id: 1,
    user: {
      id: 4,
      isPro: false,
      name: `Max`,
      avatarUrl: `img/1.png`
    },
    rating: 4,
    comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    date: `2019-05-08T14:13:56.569Z`
  }
];

export const reviewItemMock = {
  id: 1,
  user: {
    id: 4,
    isPro: false,
    name: `Max`,
    avatarUrl: `img/1.png`
  },
  rating: 4,
  comment: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  date: `2019-05-08T14:13:56.569Z`
};
