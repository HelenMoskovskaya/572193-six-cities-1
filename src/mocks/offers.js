export default [
  {
    city: `Paris`,
    centerCityCoords: [48.85881005, 2.32003101],
    offerCoords: [48.83832557, 2.29878187],
    title: `Wood and stone place`,
    type: `Private room`,
    image: `img/room.jpg`,
    price: 80,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Paris`,
    centerCityCoords: [48.85881005, 2.32003101],
    offerCoords: [48.86905515, 2.36973166],
    title: `Wood and stone place`,
    type: `Private room`,
    image: `img/room.jpg`,
    price: 80,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Cologne`,
    centerCityCoords: [50.938361, 6.959974],
    offerCoords: [50.92812784, 6.9745481],
    title: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    image: `img/apartment-03.jpg`,
    price: 180,
    rating: 100,
    isPremium: true,
  },
  {
    city: `Cologne`,
    centerCityCoords: [50.938361, 6.959974],
    offerCoords: [50.94851179, 6.9483161],
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    image: `img/apartment-02.jpg`,
    price: 132,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Brussels`,
    centerCityCoords: [50.8465573, 4.351697],
    offerCoords: [50.82894742, 4.37661409],
    title: `Wood and stone place`,
    type: `Private room`,
    image: `img/room.jpg`,
    price: 80,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Amsterdam`,
    centerCityCoords: [52.38333, 4.9],
    offerCoords: [52.3909553943508, 4.85309666406198],
    title: `Beautiful luxurious apartment at great location`,
    type: `Apartment`,
    image: `img/apartment-01.jpg`,
    price: 120,
    rating: 93,
    isPremium: true,
  },
  {
    city: `Amsterdam`,
    centerCityCoords: [52.38333, 4.9],
    offerCoords: [52.3909553943508, 4.929309666406198],
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    image: `img/apartment-02.jpg`,
    price: 132,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Hamburg`,
    centerCityCoords: [53.550341, 10.000654],
    offerCoords: [53.55453394, 10.01213372],
    title: `Beautiful luxurious apartment at great location`,
    type: `Apartment`,
    image: `img/apartment-01.jpg`,
    price: 120,
    rating: 93,
    isPremium: true,
  },
  {
    city: `Dusseldorf`,
    centerCityCoords: [51.2254018, 6.7763137],
    offerCoords: [51.22621102, 6.78346485],
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    image: `img/apartment-02.jpg`,
    price: 132,
    rating: 80,
    isPremium: false,
  },
  {
    city: `Dusseldorf`,
    centerCityCoords: [51.2254018, 6.7763137],
    offerCoords: [51.23314105, 6.77579641],
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    image: `img/apartment-02.jpg`,
    price: 132,
    rating: 80,
    isPremium: false,
  },
];
