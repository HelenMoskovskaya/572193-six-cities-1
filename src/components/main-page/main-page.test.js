import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {MainPage} from './main-page.jsx';
import {mockOffers, mockOffer, cityMock, cities, userData} from '../../mocks/mocks.js';


const renderer = new ShallowRenderer();

describe(`MainPage`, () => {
  it(`renders correctly`, () => {
    renderer.render(
        <MainPage
          offers={mockOffers}
          cities={cities}
          onCityClick={jest.fn()}
          city={cityMock}
          isAuthorizationRequired={true}
          userData={userData}
          isLoadOffers={true}
          activeItem={mockOffer}
        />
    );
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
