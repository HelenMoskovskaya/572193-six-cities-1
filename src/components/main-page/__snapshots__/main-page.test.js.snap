// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders correctly 1`] = `
<div
  className="page page--gray page--main"
>
  <Svg />
  <Header
    isAuthorizationRequired={true}
    userData={
      Object {
        "avatarUrl": "img/1.png",
        "email": "Oliver.conner@gmail.com",
        "id": 1,
        "isPro": false,
        "name": "Oliver.conner",
      }
    }
  />
  <main
    className="page__main page__main--index"
  >
    <h1
      className="visually-hidden"
    >
      Cities
    </h1>
    <CitiesList
      cities={
        Array [
          "Amsterdam",
          "Paris",
          "Colone",
        ]
      }
      city={
        Object {
          "location": Object {
            "latitude": 51.225402,
            "longitude": 6.776314,
            "zoom": 13,
          },
          "name": "Amsterdam",
        }
      }
      onCityClick={[MockFunction]}
    />
    <div
      className="cities__places-wrapper"
    >
      <div
        className="cities__places-container container"
      >
        <section
          className="cities__places places"
        >
          <h2
            className="visually-hidden"
          >
            Places
          </h2>
          <b
            className="places__found"
          >
            2 places 
            to stay in Amsterdam
          </b>
          <Memo(ConnectFunction) />
          <OfferList
            offers={
              Array [
                Object {
                  "bedrooms": 2,
                  "city": Object {
                    "location": Object {
                      "latitude": 51.225402,
                      "longitude": 6.776314,
                      "zoom": 13,
                    },
                    "name": "Amsterdam",
                  },
                  "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                  "goods": Array [
                    "Breakfast",
                    "Washer",
                    "Laptop friendly workspace",
                  ],
                  "host": Object {
                    "avatarUrl": "img/avatar-angelina.jpg",
                    "id": 25,
                    "isPro": true,
                    "name": "Angelina",
                  },
                  "id": 1,
                  "images": Array [
                    "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
                  ],
                  "isFavorite": false,
                  "isPremium": true,
                  "location": Object {
                    "latitude": 51.214402,
                    "longitude": 6.764314000000001,
                    "zoom": 16,
                  },
                  "maxAdults": 8,
                  "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                  "price": 393,
                  "rating": 2.7,
                  "title": "Waterfront with extraordinary view",
                  "type": "hotel",
                },
                Object {
                  "bedrooms": 2,
                  "city": Object {
                    "location": Object {
                      "latitude": 51.225402,
                      "longitude": 6.776314,
                      "zoom": 13,
                    },
                    "name": "Amsterdam",
                  },
                  "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                  "goods": Array [
                    "Breakfast",
                    "Washer",
                    "Laptop friendly workspace",
                  ],
                  "host": Object {
                    "avatarUrl": "img/avatar-angelina.jpg",
                    "id": 25,
                    "isPro": true,
                    "name": "Angelina",
                  },
                  "id": 3,
                  "images": Array [
                    "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
                  ],
                  "isFavorite": false,
                  "isPremium": true,
                  "location": Object {
                    "latitude": 51.214402,
                    "longitude": 6.764314000000001,
                    "zoom": 16,
                  },
                  "maxAdults": 8,
                  "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                  "price": 393,
                  "rating": 2.7,
                  "title": "Waterfront with extraordinary view",
                  "type": "hotel",
                },
              ]
            }
          />
        </section>
        <div
          className="cities__right-section"
        >
          <section
            className="cities__map map"
          >
            <MapCity
              activeOffer={
                Object {
                  "bedrooms": 2,
                  "city": Object {
                    "location": Object {
                      "latitude": 51.225402,
                      "longitude": 6.776314,
                      "zoom": 13,
                    },
                    "name": "Amsterdam",
                  },
                  "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                  "goods": Array [
                    "Breakfast",
                    "Washer",
                    "Laptop friendly workspace",
                  ],
                  "host": Object {
                    "avatarUrl": "img/avatar-angelina.jpg",
                    "id": 25,
                    "isPro": true,
                    "name": "Angelina",
                  },
                  "id": 1,
                  "images": Array [
                    "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                    "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
                  ],
                  "isFavorite": false,
                  "isPremium": true,
                  "location": Object {
                    "latitude": 51.214402,
                    "longitude": 6.764314000000001,
                    "zoom": 16,
                  },
                  "maxAdults": 8,
                  "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                  "price": 393,
                  "rating": 2.7,
                  "title": "Waterfront with extraordinary view",
                  "type": "hotel",
                }
              }
              city={
                Object {
                  "location": Object {
                    "latitude": 51.225402,
                    "longitude": 6.776314,
                    "zoom": 13,
                  },
                  "name": "Amsterdam",
                }
              }
              isZoomChange={true}
              offers={
                Array [
                  Object {
                    "bedrooms": 2,
                    "city": Object {
                      "location": Object {
                        "latitude": 51.225402,
                        "longitude": 6.776314,
                        "zoom": 13,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                    "goods": Array [
                      "Breakfast",
                      "Washer",
                      "Laptop friendly workspace",
                    ],
                    "host": Object {
                      "avatarUrl": "img/avatar-angelina.jpg",
                      "id": 25,
                      "isPro": true,
                      "name": "Angelina",
                    },
                    "id": 1,
                    "images": Array [
                      "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
                    ],
                    "isFavorite": false,
                    "isPremium": true,
                    "location": Object {
                      "latitude": 51.214402,
                      "longitude": 6.764314000000001,
                      "zoom": 16,
                    },
                    "maxAdults": 8,
                    "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                    "price": 393,
                    "rating": 2.7,
                    "title": "Waterfront with extraordinary view",
                    "type": "hotel",
                  },
                  Object {
                    "bedrooms": 2,
                    "city": Object {
                      "location": Object {
                        "latitude": 51.225402,
                        "longitude": 6.776314,
                        "zoom": 13,
                      },
                      "name": "Amsterdam",
                    },
                    "description": "I am happy to welcome you to my apartment in the city center! Three words: location, cosy and chic!",
                    "goods": Array [
                      "Breakfast",
                      "Washer",
                      "Laptop friendly workspace",
                    ],
                    "host": Object {
                      "avatarUrl": "img/avatar-angelina.jpg",
                      "id": 25,
                      "isPro": true,
                      "name": "Angelina",
                    },
                    "id": 3,
                    "images": Array [
                      "https://es31-server.appspot.com/six-cities/static/hotel/18.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/12.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/20.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/17.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/15.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/2.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/8.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/5.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/6.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/7.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
                      "https://es31-server.appspot.com/six-cities/static/hotel/14.jpg",
                    ],
                    "isFavorite": false,
                    "isPremium": true,
                    "location": Object {
                      "latitude": 51.214402,
                      "longitude": 6.764314000000001,
                      "zoom": 16,
                    },
                    "maxAdults": 8,
                    "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/1.jpg",
                    "price": 393,
                    "rating": 2.7,
                    "title": "Waterfront with extraordinary view",
                    "type": "hotel",
                  },
                ]
              }
            />
          </section>
        </div>
      </div>
    </div>
    ;
  </main>
</div>
`;
